File: object_relations/app/models/restaurant.rb
class Restaurant
  attr_accessor :name, :review_list, :customer_list

  @@all = []

  def initialize(name)
    @name = name
    @review_list = []
    @customer_list = []
    @@all << self
  end

  def reviews
    @review_list
  end

  def customers
    @customer_list
  end

  def self.all
    @@all
  end

  def self.find_by_name(name)
    Restaurant.all.find { |restaurant| restaurant.name == name }
  end

end

File: javascript/models/restaurant.js
var Restaurant = (function createRestaurant() {
  var id = 0;
  return class Restaurant {
    constructor(name) {
      this.name = name;
      this.id = ++id;
      this.customers = [];
      this.reviews = [];
      store.restaurants = [...store.restaurants, this]
    }

    static all() {
      return store.restaurants
    }

    static find_by_name(name) {
      return store.restaurants.find((restaurant) => { return restaurant.name === name })
    }
  }
})()


Restaurant.prototype.customers = () => {
  return this.customers;
}

Restaurant.prototype.reviews = () => {
  return this.reviews;
}
